hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -426,9 +426,9 @@ public void run(Packet packet) throws Exception {
      * <p>
      * It makes an assumption that the callId is the first long field in the serialized operation.
      */
-    private long extractOperationCallId(Data data) throws IOException {
+    private long extractOperationCallId(Packet packet) throws IOException {
         ObjectDataInput input = ((SerializationServiceV1) node.getSerializationService())
-                .initDataSerializableInputAndSkipTheHeader(data);
+                .initDataSerializableInputAndSkipTheHeader(new HeapData(packet.getPayload()));
```

Comment:
This looks like a good candidate for a `Packet::getCallId()` method to skip the whole serialization part. This would also get rid of the introduced `new HeapData()`.